packages/apollo-link-state/src/index.ts

```diff
@@ -53,10 +53,8 @@ export const withClientState = (
       if (fieldValue !== undefined) return fieldValue;
 
       // Look for the field in the custom resolver map
-      const resolve =
-        resolvers[(rootValue as any).__typename || type][
-          info.resultKey || fieldName
-        ];
+      const resolverMap = resolvers[(rootValue as any).__typename || type];
+      const resolve = resolverMap[fieldName] || resolverMap[info.resultKey];
```

Comment:
It's possible we shouldn't fall back to using the `info.resultKey` to look up a resolver, at all? /question for someone like @peggyrayzis?

Reply:
You're right, I think we can delete that fallback.

I also spot another possible bug w/ aliases and default resolvers on line 52 above. 🐛